dynamo-future-file-extension/extension/Extension.cs

```diff
@@ -0,0 +1,128 @@
+﻿using Dynamo.Extensions;
+using Dynamo.Graph.Nodes;
+using Dynamo.Models;
+using HarmonyLib;
+using Newtonsoft.Json.Linq;
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Serialization;
+using System.Xml;
+using Dynamo.Logging;
+using System.Diagnostics;
+
+namespace Dynamo.FutureFileExtension
+{
+    public class FutureFileExtension : IExtension, ILogSource
+    {
+        private static MethodInfo openxmlmethod;
+        private static MethodInfo openJsonFileMethod;
+
+        public event Action<ILogMessage> MessageLogged;
+
+        public string UniqueId => "277edc98-d327-404e-979c-7c6e6f6956b2";
+
+        public string Name => "FutureFileExtension_File_Compatability";
+
+        public void Dispose()
+        {
+
+        }
+
+        private static string ToEnumString<T>(T type)
+        {
+            var enumType = typeof(T);
+            var name = Enum.GetName(enumType, type);
+            var enumMemberAttribute = ((EnumMemberAttribute[])enumType.GetField(name).GetCustomAttributes(typeof(EnumMemberAttribute), true)).Single();
+            return enumMemberAttribute.Value;
+        }
+
+        public void Ready(ReadyParams sp)
+        {
+            //Debugger.Launch();
+            //if dynamo is newer than 2.11, don't monkey patch anything.
+            if (sp.StartupParams.DynamoVersion > new Version(2, 12, 0))
+            {
+                MessageLogged(LogMessage.Info($"This extension only patches dynamo versions older than 2.12, current version is {Assembly.GetCallingAssembly().GetName().Version}, returning."));
+                return;
+            }
+
+            DoPatching();
+        }
+
+        public void DoPatching()
+        {
+            var harmony = new Harmony("Dynamo.FurureFileExtension.Patch.1");
+
+            var originalOpenFileFromPath = AccessTools.Method(typeof(DynamoModel), nameof(DynamoModel.OpenFileFromPath));
+            var prefix = typeof(FutureFileExtension).GetMethod(nameof(Prefix), System.Reflection.BindingFlags.Static | System.Reflection.BindingFlags.Public);
+            openxmlmethod = typeof(DynamoModel).GetMethod("OpenXmlFileFromPath", System.Reflection.BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
Can you use `nameof(DynamoModel.OpenXmlFileFromPath)` here?

Reply:
unfortunately, no because it's private.